cdc/sinkv2/roweventsink/row_event_sink.go

```diff
@@ -24,16 +29,121 @@ import (
 // determine if we really need to process the event.
 type RowEvent struct {
 	Row          *model.RowChangedEvent
-	Callback     func()
 	TableStopped *atomic.Bool
+    partitionedProgress *tablesink.PartitionedProgress
 }
 
 // RowEventSink is a sink that processes row events.
 // Usually, it is a MQ sink or S3 sink(not implemented).
 type RowEventSink interface {
 	// WriteRowChangedEvents writes row changed events to the sink.
 	// Note: This is an asynchronous and thread-safe method.
+    // postFlush should be thread-safe.
 	WriteRowChangedEvents(rows ...*RowEvent) error
 	// Close closes the sink.
 	Close() error
 }
+
+type rowEventSinkImpl struct {
+    input chan<-*RowEvent
+    eventRouter *dispatcher.EventRouter
+	topicManager         manager.TopicManager
+	flushWorker          *flushWorker
+}
+
+func (r *rowEventSinkImpl) WriteRowChangedEvents(rows ...*RowEvent) error {
+    for _, row := range rows {
+        r.input <- rows
```

Comment:
Can this block?

Reply:
Nice catch, should use an unbounded channel.